multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/cf/clients/CustomControllerClient.java

```diff
@@ -58,8 +59,8 @@ private String addPageOfResources(String path, List<Map<String, Object>> allReso
         }
         if (responseMap.containsKey("included") && includedResources != null) {
             @SuppressWarnings("unchecked")
-            Map<String, Object> included = (Map<String, Object>) responseMap.get("included");
-            includedResources.putAll(included);
+            var included = (Map<String, List<Map<String, Object>>>) responseMap.get("included");
+            included.forEach((key, resources) -> includedResources.merge(key, resources, ListUtils::union));
```

Comment:
Because previously, the included resources would have been only those from the last page of resources (it was overwriting the entries).
Included works a bit iffy in that it consists of the same keys every time but the values may be different depending on the page of resources. That's why we merge the included resources instead of overwriting them.